src/libraries/System.Private.CoreLib/src/System/Decimal.cs

```diff
@@ -102,41 +102,37 @@ namespace System
         // and finally bit 31 indicates the sign of the Decimal value, 0 meaning
         // positive and 1 meaning negative.
         //
-        // NOTE: Do not change the order in which these fields are declared. The
-        // native methods in this class rely on this particular order.
-        // Do not rename (binary serialization).
-        private readonly int flags;
-        private readonly int hi;
-        private readonly int lo;
-        private readonly int mid;
+        // NOTE: Do not change the order and types of these fields. The layout has to
+        // match Win32 DECIMAL type.
+        private readonly int _flags;
+        private readonly uint _hi32;
+        private readonly ulong _lo64;
 
         // Constructs a Decimal from an integer value.
         //
         public Decimal(int value)
```

Comment:
Begin 00007FFE9E193740, size 1e
```
00007ffe`9e193740 85d2            test    edx,edx
00007ffe`9e193742 7c06            jl      System_Private_CoreLib!System.Decimal..ctor(Int32)+0xffffffff`a273ea2a (00007ffe`9e19374a)
00007ffe`9e193744 33c0            xor     eax,eax
00007ffe`9e193746 8901            mov     dword ptr [rcx],eax
00007ffe`9e193748 eb08            jmp     System_Private_CoreLib!System.Decimal..ctor(Int32)+0xffffffff`a273ea32 (00007ffe`9e193752)
00007ffe`9e19374a c70100000080    mov     dword ptr [rcx],80000000h
00007ffe`9e193750 f7da            neg     edx
00007ffe`9e193752 895108          mov     dword ptr [rcx+8],edx
00007ffe`9e193755 33c0            xor     eax,eax
00007ffe`9e193757 89410c          mov     dword ptr [rcx+0Ch],eax
00007ffe`9e19375a 894104          mov     dword ptr [rcx+4],eax
00007ffe`9e19375d c3              ret
```

After:
- One less memory access
- Instruction count and code size is the same
```
Begin 00007FFE94E4B6B0, size 1e
00007ffe`94e4b6b0 85d2            test    edx,edx
00007ffe`94e4b6b2 7c06            jl      System_Private_CoreLib!System.Decimal..ctor(Int32)+0x6a (00007ffe`94e4b6ba)
00007ffe`94e4b6b4 33c0            xor     eax,eax
00007ffe`94e4b6b6 8901            mov     dword ptr [rcx],eax
00007ffe`94e4b6b8 eb08            jmp     System_Private_CoreLib!System.Decimal..ctor(Int32)+0x72 (00007ffe`94e4b6c2)
00007ffe`94e4b6ba c70100000080    mov     dword ptr [rcx],80000000h
00007ffe`94e4b6c0 f7da            neg     edx
00007ffe`94e4b6c2 8bc2            mov     eax,edx
00007ffe`94e4b6c4 48894108        mov     qword ptr [rcx+8],rax
00007ffe`94e4b6c8 33c0            xor     eax,eax
00007ffe`94e4b6ca 894104          mov     dword ptr [rcx+4],eax
00007ffe`94e4b6cd c3              ret
```